### ISSUE TITLE
can't select a given form with br.select_form(action='target_page_url')

### ISSUE THREAD

**Issue by abubelinha** [python-mechanize/mechanize#78]
Maybe I am misunderstanding something. Please correct me if I am wrong:

**Code:**

```Python
import mechanize
br = mechanize.Browser()
url = "http://example.org/"
br.open(url+"resource.do?r=myresource")
# I output details of all forms in current page, so I can know how to select the one I want:
myforms = br.forms() 
for f in myforms:
    print(f.name, f.action, f.attrs)
# now I try to select the one I know, using the action property, but THIS FAILS:
myforms = br.select_form(action = url + "manage/addsource.do")
```
**Output:** (I just show the last forms in the loop)

```
*** name:  None
    action: http://example.org/manage/addsource.do
    attrs:  {'action': 'addsource.do', 'method': 'post', 'enctype': 'multipart/form-data'}

 <post http://example.org/manage/addsource.do multipart/form-data
  <HiddenControl(r=test-occ) (readonly)>
  <HiddenControl(validate=false) (readonly)>
  <SelectControl(sourceType=[(*), source-sql, source-file, source-url])>
  <FileControl(file=<No files added>)>
  <TextControl(sourceName=)>
  <TextControl(url=)>
  <SubmitControl(add=Connect) (readonly)>
  <SubmitControl(clear=Clear) (readonly)>>

*** name:  None
    action: http://example.org/manage/canceloverwrite.do
    attrs:  {'action': 'canceloverwrite.do', 'method': 'post'}

 <post http://example.org/manage/canceloverwrite.do application/x-www-form-urlencoded
  <HiddenControl(r=test-occ) (readonly)>
  <HiddenControl(validate=false) (readonly)>
  <SubmitControl(canceloverwrite=Cancel) (readonly)>>

Traceback (most recent call last):
  File "webscrapping_tests.py", line 701, in <module>
    mechanize_test(url=loginfo["urlbase"], user=loginfo["username"], pwd=loginfo["password"])
  File "webscrapping_tests.py", line 678, in mechanize_test
    myforms = br.select_form(action = url+"/manage/addsource.do")
  File "C:\Python38\lib\site-packages\mechanize\_mechanize.py", line 681, in select_form
    raise FormNotFoundError("no form matching " + description)
mechanize._mechanize.FormNotFoundError: no form matching action = 'http://example.org/manage/addsource.do'
```

What am I doing wrong in the last line of my code?
I am trying to replicate the first line in the example code section under [`select_form()` documentation](https://mechanize.readthedocs.io/en/latest/browser_api.html#mechanize.Browser.select_form) 

As for the output of the previous loop, that form exists in that page.
I tried also with `select_form(action = 'addsource.do')` but it didn't work either.

Thanks in advance for any help

**Comment by kovidgoyal** [comment]
the action is addsource.do not url + /manage _ addsource.do

**Comment by abubelinha** [comment]
@kovidgoyal   Would you mind to reopen this issue?
I can see any reason for it to be closed

**Comment by kovidgoyal** [comment]
Then you arent doing something you think you are doing. You need to provide a complete reproducible failure case.

**Comment by abubelinha** [comment]
OK, I'll try.
In the mean time, why don't you keep this open?
There is not a complete reproducible success case in your answer, nor in documentation or examples section.

Actually, you say "the action is addsource.do not url + /manage _ addsource.do", whereas documentation shows an example with a full url. Can you explain why?

Thanks

**Comment by kovidgoyal** [comment]
I've spent about as much time helping you as I feel like. And
sure I will re-open the ticket since there appears to be a section of
the population that takes the status of tickets as a personal insult.
Enjoy the sounds of crickets. Good luck.

**Comment by abubelinha** [comment]
Thanks.  I also spent time trying to show my problem.  
Closing it is not an insult, but it's not very polite to close issues right away, not asking authors if provided solutions work.

You need a login to show the page forms I reproduced, so I'll try to generate a public static web page with the same content and a full .py script trying to read that form. 
Please give me a couple of days. If I can't make this reproducible I'll close it myself.

But I keep not understanding why you see something wrong in putting the whole action url in my code: the only example in documentation shows an example with a full url (which should be the case if your form submits info to a different server).

**Comment by abubelinha** [comment]
I think I got a reproducible example.

In I try of making it useful for everybody, I forgot about my own web page and attempted some more general tests based on the same website you have in the pypi.py example.

I posted it in #79 
Hopefully when that gets closed this can be closed as well.

Thanks a lot for your time

**Comment by abubelinha** [comment]
Solved by #79
